modules/imgcodecs/src/loadsave.cpp

```diff
@@ -462,16 +462,10 @@ imread_( const String& filename, int flags, OutputArray mat )
     // grab the decoded type
     const int type = calcType(decoder->type(), flags);
 
-    if (mat.empty())
```

Comment:
this change seems wrong to me.

Reply:
`create` should not reallocate data if requested image size fits into existing Mat.

```.cpp
Mat img(1000, 1000, CV_8UC3);
imread(f, img); // should not reallocate data if image size is less than (1000, 1000)
```

Without this change it will not be possible to read multiple images of different sizes efficiently, because user would need to pass either empty Mat or a Mat with exact image dimensions on each iteration.

I.e. with proposed change, following code should work and should perform reallocations only when necessary:
```.cpp
Mat img;
for (auto f : filenames)
{
    imread(f, img); // will fail if images have different sizes
}
```

Reply:
sorry you are right. `mat.create( size.height, size.width, type );` does the job.

Reply:
```
    TickMeter tm;

    Mat im;

    for (int i = 0; i < 5; i++)
    {
        tm.reset();
        tm.start();
        imread("read.png", im);
        tm.stop();
        printf("read time %f\n", tm.getTimeSec());
    }
    imshow("read.png", im);
    waitKey();
```

```
read time 0.692092
read time 0.615938
read time 0.615163
read time 0.615883
read time 0.617650
```
good job!